### ISSUE TITLE
Uncaught TypeError: Cannot read property 'nodeType' of null in shared.js stripCommentsFromElement(element)

### ISSUE THREAD

**Issue by akuznetsov-gridgain** [angular/angular.js#14211]
Hi!

It seems than function "stripCommentsFromElement"  in line 148 is missing check for null.

```javascript
 if (element.nodeType === ELEMENT_NODE) {
    return jqLite(element);
  }
```

I got this exception in following case.
I have a list of items with "x" button that have a tooltip animated by angular.
So when user place mouse over "x" button a tooltip is shown.
User click on "x" - line deleted and tooltip is trying to hide,  but failed with "Cannot read property".

My stacktrace:

```
Uncaught TypeError: Cannot read property 'nodeType' of null stripCommentsFromElement 
 @ angular-animate.js:154queueAnimation
 @ angular-animate.js:2425push
 @ angular-animate.js:2380leave 
 @ angular.js:5378TooltipFactory.$tooltip.hide 
 @ angular-strap.js:204(anonymous function) 
 @ angular-strap.js:190
```

**Comment by Narretz** [comment]
Can you please post a runnable example in a https://plnkr.co or similar?`Without that, it's difficult to say where the problem is and we also can't write a test for it.

**Comment by Narretz** [comment]
Since you are using angular-strap, you should also open an issue there as it might be an issue with the library.

**Comment by Narretz** [comment]
I'm going to close this issue because we haven't got any feedback. Feel free to reopen this issue if you can provide new feedback.

**Comment by joshwhatk** [comment]
I am experiencing this same problem in my code, and have been able to reproduce it here: [plnkr.co/edit/ljytGHcKN1s7aKXNrWZg?p=preview](http://plnkr.co/edit/ljytGHcKN1s7aKXNrWZg?p=preview)

It was found in some old code while migrating to newer versions of Angular, AngularStrap, etc. I may be simply handling objects in the wrong way, in which case, any guidance would be much appreciated.

**Comment by gkalpak** [comment]
This is actually a bug in angular-strap (but one that is only triggered by a certain combination of events).
Here is (the simplified version of) what happens:

1. You click the "Save" button.
2. In your callback, you create a new object for your `objects` array.
3. `ngRepeat` detects the change and destroys the current row.
4. As part of destroying the row, all child-scopes of the current row's scope get destroyed (inluding the modal's scope).
5. As a reaction to the `$destroy` event on it's scope, the modal removes itself from the DOM and an internal `modalInstance` reference is set to null to help GC.
6. At the end of your "Save" callback, you call `$hide()`, which schedules to remove the modal.
7. The removal is scheduled using the (now nullified) `modalInstance` variable.
8. Everyone knows what happens next.

So, basically, it something that nees to be fixed n angular-strap. (While we could still add some safe-guards in Angular/ngAnimate, this would only hide the error and cover an existing problem, not make it go away - so I'm against this.)

--
As a temporary&trade; workaround, you could avoid deleting a row after each edit, either by tracking your repeated objects by index (or some other property that make sense for your usecase), or re-using the same object instance.

[Updated plnkr 1][1]
[Updated plnkr 2][2]

[1]: http://plnkr.co/edit/ljytGHcKN1s7aKXNrWZg?p=preview
[2]: http://plnkr.co/edit/iCSWyQJkz2l0lThaqI0D?p=preview

**Comment by joshwhatk** [comment]
Cool, works for me. I'll try making an issue with AngularStrap and look for ways to avoid `$destroy`ing the scope before the modal is closed.

Thanks!

**Comment by akuznetsov-gridgain** [comment]
I can confirm that it is a bug in AngularStrap $tooltip (in my case)
I opened an issue in AngularStrap github
https://github.com/mgcrea/angular-strap/issues/2090

We workaround by creating a delegate and replacing $tooltip.hide function with patched version.

You may go to AngularStrap issues and vote for fix by +1.

**Comment by mohiniyadav** [comment]
Hi,
I am getting this error. Can you tell me how can i resolve this.

Uncaught TypeError: Cannot read property 'nodeType' of null
    at a (fusioncharts.js?v=${prefix.revision}:59)
    at a (fusioncharts.js?v=${prefix.revision}:60)
    at a (fusioncharts.js?v=${prefix.revision}:60)
    at c (fusioncharts.js?v=${prefix.revision}:60)
